app/assets/javascripts/select-kit/components/user-chooser.js.es6

```diff
@@ -0,0 +1,85 @@
+import MultiSelectComponent from "select-kit/components/multi-select";
+import { computed } from "@ember/object";
+import {
+  default as userSearch,
+  skipSearch,
+  eagerCompleteSearch
+} from "discourse/lib/user-search";
+import { makeArray } from "discourse-common/lib/helpers";
+
+export default MultiSelectComponent.extend({
+  pluginApiIdentifiers: ["user-chooser"],
+  classNames: ["user-chooser"],
+  valueProperty: "username",
+
+  modifyComponentForRow() {
+    return "user-chooser/user-row";
+  },
+
+  selectKitOptions: {
+    topicId: undefined,
+    categoryId: undefined,
+    includeGroups: false,
+    allowedUsers: false,
+    includeMentionableGroups: false,
+    includeMessageableGroups: false,
+    allowEmails: false,
+    groupMembersOf: undefined
+  },
+
+  content: computed("value.[]", function() {
+    return Ember.makeArray(this.value).map(x => this.defaultItem(x, x));
+  }),
+
+  excludedUsers: computed(
+    "value",
+    "currentUser",
+    "selectKit.options.{excludeCurrentUser,excludedUsernames}",
+    {
+      get() {
+        const options = this.selectKit.options;
+        let usernames = makeArray(this.value);
+
+        if (this.currentUser && options.excludeCurrentUser) {
+          usernames.concat([this.currentUser.username]);
+        }
+
+        return usernames.concat(options.excludedUsernames || []);
+      }
+    }
+  ),
+
+  search(filter = "") {
+    filter = filter || "";
+    const options = this.selectKit.options;
+
+    // prevents doing ajax request for nothing
+    const skippedSearch = skipSearch(filter, options.allowEmails);
+    const eagerComplete = eagerCompleteSearch(
+      filter,
+      options.topicId || options.categoryId
+    );
+    if (skippedSearch || (filter === "" && !eagerComplete)) {
+      return;
+    }
+
+    return userSearch({
+      term: filter,
+      topicId: options.topicId,
+      categoryId: options.categoryId,
+      exclude: this.excludedUsers,
+      includeGroups: options.includeGroups,
+      allowedUsers: options.allowedUsers,
+      includeMentionableGroups: options.includeMentionableGroups,
+      includeMessageableGroups: options.includeMessageableGroups,
+      groupMembersOf: options.groupMembersOf,
+      allowEmails: options.allowEmails
+    }).then(result => {
+      if (typeof result === "string") {
+        // do nothing promise probably got cancelled
```

Comment:
This is a bit odd - you saw this error in testing?

Reply:
Not that's not odd, this is fully expected. I guess this is an artifact of the past. But when we cancel userSearch will return some _CANCELED string.

Promise cancelation can happen if you have an empty filter and no topicId/categoryId,  these files (userSearch, category tag search...) would need a huge refactoring, but didn't want it touch it for now. So I handled this case like this for now.

Reply:
So it's our code that produces _CANCELLED? In that case this is OK but it's just an odd pattern to have to check for a string as an API. If it's not new here let's leave it for now.